src/components/timeline/TimelineHistogram.svelte

```diff
@@ -313,7 +315,12 @@
       timelineHovering = true;
       cursorLeft = clamp(x - histRect.left, 0, histRect.width); // Ensure cursor is within range
       const cursorTime = xScaleMax.invert(cursorLeft);
-      cursorTooltip = getDoyTime(cursorTime, false);
+      cursorTooltip = $plugins.time.primary.format(cursorTime) ?? 'Invalid Date';
+
+      // Remove milliseconds if DOY-like time
+      if (validateTime(cursorTooltip, TimeTypes.ABSOLUTE)) {
```

Comment:
Should this be moved into a function for re-use?

Reply:
Good idea, done